volatility3/framework/plugins/linux/psscan.py

```diff
@@ -0,0 +1,127 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+from typing import Iterable, List, Tuple
+import struct
+
+from volatility3.framework import renderers, interfaces, symbols, constants
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.layers import scanners
+from volatility3.framework.renderers import format_hints
+
+
+class PsScan(interfaces.plugins.PluginInterface):
+    """Scans for processes present in a particular linux image."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    def _get_task_fields(
+        self, task: interfaces.objects.ObjectInterface
+    ) -> Tuple[int, int, int, str]:
+        """Extract the fields needed for the final output
+
+        Args:
+            task: A task object from where to get the fields.
+        Returns:
+            A tuple with the fields to show in the plugin output.
+        """
+        pid = task.tgid
```

Comment:
These seemed to be named... confusingly.  I don't know the linux structures well enough, so I'm going to assume this is right, but just wanted you to check to make sure it wasn't a typo please.  5:)

Reply:
Yes it is weird, its to match the pslist plugin (https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/plugins/linux/pslist.py#L83) . 

It's the way the kernel replies to the getpid() function itself - https://github.com/torvalds/linux/blob/master/kernel/sys.c#L932. 

This stackoverflow reply has a nice explanation too - https://stackoverflow.com/questions/9305992/if-threads-share-the-same-pid-how-can-they-be-identified